lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -637,7 +637,12 @@ struct RvalueExprVisitor {
         cond = builder.create<moore::AndOp>(loc, leftValue, rightValue);
       } else {
         // Handle expressions.
-        if (!listExpr->type->isSimpleBitVector()) {
+        auto ty = listExpr->type;
+        bool isUnPackedStruct = ty->isUnpackedStruct();
+        bool isUnPackedArray = ty->isUnpackedArray();
+        bool isUnPackedUnion = ty->isUnpackedUnion();
+
+        if (isUnPackedArray || isUnPackedStruct || isUnPackedUnion) {
```

Comment:
I would use something like `!ty->isPacked()` here, instead of listing concrete unpacked types. For example, things like a `string` is also unpacked, which would be missed here. It's often easier to use a whitelist, i.e. error out if the type isn't strictly what you support, instead of using a blacklist, i.e. making a list of all the things that are not supported.

Reply:
Oh yeah, I hadn't realized that `isPacked()` doesn't exist. Browsing through `slang/ast/types/Type.h` a bit, it looks like `isIntegral()` is a way to determine if the type is essentially packed:
```c++
    /// Indicates whether this is an integral type, which includes all scalar types, predefined
    /// integer types, packed arrays, packed structures, packed unions, and enum types.
    bool isIntegral() const;
```
Maybe something like 
```c++
if (!ty->isIntegral()) {
  // ...
}
```
could work in your code?

Reply:
That will work, thanks! 💯